src/plugins/controls/server/control_types/options_list/options_list_queries.ts

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { get, isEmpty } from 'lodash';
+
+import { SearchResponse } from '@elastic/elasticsearch/lib/api/types';
+import { getFieldSubtypeNested } from '@kbn/data-views-plugin/common';
+
+import { OptionsListRequestBody } from '../../../common/control_types/options_list/types';
+
+export interface OptionsListAggregationBuilder {
+  buildAggregation: (req: OptionsListRequestBody) => unknown;
+  parse: (response: SearchResponse) => string[];
+}
+
+/**
+ * Validation aggregations
+ */
+export const getValidationAggregationBuilder: () => OptionsListAggregationBuilder = () => ({
+  buildAggregation: ({ selectedOptions, fieldName }: OptionsListRequestBody) => {
+    const selectedOptionsFilters = selectedOptions?.reduce((acc, currentOption) => {
+      acc[currentOption] = { match: { [fieldName]: currentOption } };
+      return acc;
+    }, {} as { [key: string]: { match: { [key: string]: string } } });
+
+    return selectedOptionsFilters && !isEmpty(selectedOptionsFilters)
+      ? {
+          filters: {
+            filters: selectedOptionsFilters,
+          },
+        }
+      : undefined;
+  },
+  parse: (rawEsResult) => {
+    const rawInvalidSuggestions = get(rawEsResult, 'aggregations.validation.buckets') as {
+      [key: string]: { doc_count: number };
+    };
+    return rawInvalidSuggestions && !isEmpty(rawInvalidSuggestions)
+      ? Object.entries(rawInvalidSuggestions)
+          ?.filter(([, value]) => value?.doc_count === 0)
+          ?.map(([key]) => key)
+      : [];
+  },
+});
+
+/**
+ * Suggestion aggregations
+ */
+export const getSuggestionAggregationBuilder = ({
+  fieldSpec,
+  textFieldName,
+  searchString,
+}: OptionsListRequestBody) => {
+  if (textFieldName && fieldSpec?.aggregatable && searchString) {
+    return suggestionAggSubtypes.keywordAndText;
+  }
+  if (fieldSpec?.type === 'boolean') {
+    return suggestionAggSubtypes.boolean;
+  }
+  if (fieldSpec && getFieldSubtypeNested(fieldSpec)) {
+    return suggestionAggSubtypes.subtypeNested;
+  }
+  return suggestionAggSubtypes.keywordOnly;
+};
+
+const getEscapedQuery = (q: string = '') =>
+  q.replace(/[.?+*|{}[\]()"\\#@&<>~]/g, (match) => `\\${match}`);
+
+const suggestionAggSubtypes: { [key: string]: OptionsListAggregationBuilder } = {
+  /**
+   * the "Keyword only" query / parser should be used when the options list is built on a field which has only keyword mappings.
+   */
+  keywordOnly: {
```

Comment:
Tested this using the `customer_gender` field in the ecommerce sample data - as expected, search was not case sensitive and, when searching for `fe`, no options showed up; but when searching for `FE`, the `FEMALE` option showed up at the top of the list.